src/django_rich/runner.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import sys
+import unittest
+from unittest.result import STDERR_LINE, STDOUT_LINE
+
+from django.test import testcases
+from django.test.runner import DebugSQLTextTestResult, DiscoverRunner, PDBDebugResult
+from rich.color import Color
+from rich.console import Console
+from rich.rule import Rule
+from rich.style import Style
+from rich.traceback import Traceback
+
+
+class RichMixin(unittest.TextTestResult):
+    # Traceback width defaults to 100
+    console = Console(width=100, stderr=True)
+    django_green = Style(color=Color.from_rgb(32, 170, 118))
+
+    def addError(self, test, err):
+        super(unittest.TextTestResult, self).addError(test, err)
+        if self.showAll:
+            self.console.print("ERROR", style=Style(color="red"))
+        elif self.dots:
+            self.console.print("E", style=Style(color="red"), end="")
+
+    def addSuccess(self, test):
+        super(unittest.TextTestResult, self).addSuccess(test)
+        if self.showAll:
+            self.console.print("ok", style=self.django_green)
+        elif self.dots:
+            self.console.print(".", style=self.django_green, end="")
+
+    def addFailure(self, test, err):
+        super(unittest.TextTestResult, self).addFailure(test, err)
+        if self.showAll:
+            self.console.print("FAIL", style=Style(color="red"))
+        elif self.dots:
+            self.console.print("F", style=Style(color="red"), end="")
+
+    def addSkip(self, test, reason):
+        super(unittest.TextTestResult, self).addSkip(test, reason)
+        if self.showAll:
+            self.console.print(f"skipped {reason!r}", style=Style(color="yellow"))
+        elif self.dots:
+            self.console.print("s", style=Style(color="yellow"), end="")
+
+    def addExpectedFailure(self, test, err):
+        super(unittest.TextTestResult, self).addExpectedFailure(test, err)
+        if self.showAll:
+            self.console.print("expected failure", style=Style(color="yellow"))
+        elif self.dots:
+            self.console.print("x", style=Style(color="yellow"), end="")
+
+    def addUnexpectedSuccess(self, test):
+        super(unittest.TextTestResult, self).addUnexpectedSuccess(test)
+        if self.showAll:
+            self.console.print("unexpected success", style=Style(color="red"))
+        elif self.dots:
+            self.console.print("u", style=Style(color="red"), end="")
+
+    def _exc_info_to_string(self, err, test):
```

Comment:
So I guess it could "just" be this function.

But while in Rome...

Reply:
I think it's good to add all the other functions to use colourization